classes/socket.h

```diff
@@ -414,9 +414,14 @@ PHP_METHOD(Socket, strerror) {
 #ifdef HAVE_SHUTDOWN
 /* {{{ proto bool Socket::shutdown(int how) */
 PHP_METHOD(Socket, shutdown) {
-	zend_long how_shutdown = 0;
+	zend_long how_shutdown = SHUT_RDWR;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "l", &how_shutdown) != SUCCESS) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|l", &how_shutdown) != SUCCESS) {
+		RETURN_FALSE;
+	}
+
+	if (how_shutdown != SHUT_RD && how_shutdown != SHUT_WR && how_shutdown != SHUT_RDWR) {
+		zend_throw_exception(spl_ce_InvalidArgumentException, "Invalid shutdown type", 0);
 		RETURN_FALSE;
```

Comment:
Since an exception is being thrown, we can change `RETURN_FALSE` to simply be a `return;`, since the return value won't be usable after an exception has been thrown.